qiskit/transpiler/passes/scheduling/measurereschedule.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Rescheduling measurements to be compatible with the meas_map constraint."""
```

Comment:
We're moving towards removing this constraint on the backend. I don't like seeing it propagating further up the stack. It's also really specific to a backend, and specific to IBM backends (as a single current restraint). I'd rather see an error message somewhere else for the time being. @ajavadia ?

Reply:
I'm guessing this could also be simplified (wherever you felt like it was needed) by setting `meas_map = [range(num_qubits)]`, which _is_ what the constraint is on all backends that I've seen, and would satisfy any other `meas_map`.

Reply:
I decided to give up to cut off the code for rescheduling to satisfy meas_map constraints as MeasureReschedule pass, eventually. 
I'll remove the MeasureReschedule class and leave the code in schedule() as it is in this PR as you suggested.
I found it's much more difficult to complete the cut-off than I expected, i.e. it will require several PRs before that. 
- We need to introduce a special measure instruction that has no clbit to store the result. That may cause bugs in many places. At least, I found the current text drawer does not work for such a measure.
- We need to make instruction_schedule_map can return qubit-wise measurement schedule. The schedule should be parameterized so that we can specify the mem_slot tied with the clbit. 
etc.

Reply:
#4639 seems including the MeasureReschedule pass idea.